cloudinit/config/cc_growpart.py

```diff
@@ -210,11 +210,35 @@ def get_size(filename):
 
 def device_part_info(devpath):
     # convert an entry in /dev/ to parent disk and partition number
+    myenv = os.environ.copy()
+    myenv['LANG'] = 'C'
 
     # input of /dev/vdb or /dev/disk/by-label/foo
     # rpath is hopefully a real-ish path in /dev (vda, sdb..)
     rpath = os.path.realpath(devpath)
 
+    is_lvm = True if re.search('.*dm.*', rpath) else False
+    if is_lvm:
```

Comment:
I think the whole block of code between 222 and 241 would work well as a function and allow for specific testing of it.

```python

func get_pvs_for_lv(devpath):
    """return a list of pvs that compose the logical volume 'devpath'
    ...
    return [pvs]